lms/djangoapps/course_wiki/middleware.py

```diff
@@ -75,6 +76,13 @@ def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: d
                     # if a user is logged in, but not authorized to see a page,
                     # we'll redirect them to the course about page
                     return redirect('about_course', course_id.to_deprecated_string())
+
+                # If enterprise integration is enabled, and course_id requires data sharing consent,
```

Comment:
i'm also wondering if we should perform the `consent_needed_for_course` check here, or if it should be folded into `get_course_specific_consent_url`, which would yield something along the lines of:

```
consent_url = get_course_specific_consent_url(request, course_id)  # Do the consent_needed_for_course inside of this
if consent_url:
    return redirect(consent_url)
```